core/trino-main/src/main/java/io/trino/execution/TaskInfo.java

```diff
@@ -116,6 +116,11 @@ public TaskInfo summarize()
         return new TaskInfo(taskStatus, lastHeartbeat, outputBuffers.summarize(), noMoreSplits, stats.summarize(), estimatedMemory, needsPlan);
     }
 
+    public TaskInfo pruneSpoolingOutputStats()
```

Comment:
This is needed to drop the spooling output statistics from the TaskInfo stored on coordinator. TaskInfo is retained in memory for every task even after query finishes (up to 100 queries in history by default). I wanted to avoid extra memory overhead, as a single spooling output stats object is about 2KB large (assuming 1000 output partitions).